src/hypothesis/settings.py

```diff
@@ -257,8 +259,41 @@ def __exit__(self, *args, **kwargs):
         default_context_manager = self.defaults_stack().pop()
         return default_context_manager.__exit__(*args, **kwargs)
 
-    default = DefaultSettings()
+    @staticmethod
+    def register_profile(name, **kwargs):
+        """
+        registers a collection of values to be used as a
+        settings profile. These settings can be loaded
+        in by name. Enable different defaults for different settings
+
+        If a profile contains an invalid setting an AttributeError will be thrown
+        """
+        for setting in kwargs.keys():
+            getattr(Settings.default, setting)
+        Settings._profiles[name] = kwargs
+
+    @staticmethod
+    def _reset_defaults():
+        for name, setting in all_settings.items():
+            setattr(Settings.default, name, setting.default)
 
+    @staticmethod
+    def load_profile(name):
```

Comment:
Minor enhancement request: It would be nice to have a Settings.get_profile or similar method as well as one that loads it in by the default, so you could explicitly use a particular profile for one particular test.